File: Data Structures & Algorithms/missing-number/submission-1.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        unordered_map<int, int> mp;
        for(int i = 0; i <= nums.size()-1; ++i){
            mp[nums[i]]++;
        }
        
        for(int i = 0; i <= nums.size(); ++i){
            if(mp.count(i) == 0){
                return i; //
            }
        }
    }
};

File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, s: List[int], val: int) -> int:
        l = 0
        count = 0
        for r in range(len(s)):
            if s[r] != val:
                count +=1
                s[l], s[r] = s[r], s[l]
                l+=1
            r+=1
        s = s[:count]
        return len(s)
        